packages/commercetools/theme/components/StoreLocaleSelector.vue

```diff
@@ -0,0 +1,177 @@
+      @click:close="isLangModalOpen = !isLangModalOpen">
+      <SfList>
+        <SfListItem v-for="store in availableStores" :key="store.id">
+          <a
+            href="javascript:void(0)"
+            class="container__store--link"
+            :class="isStoreSelected(store) ? 'container__store--selected' : ''"
+            @click="changeStore(store)"
+          >
+            <SfCharacteristic class="language">
+              <template #title>
+                <span>{{ store.name }}</span>
+              </template>
+              <template #icon>
+                <SfImage :src="`/icons/langs/${getStoreLocale(store)}.webp`" width="20" alt="Flag" class="language__flag" />
+              </template>
+            </SfCharacteristic>
+          </a>
+        </SfListItem>
+      </SfList>
+
+      <SfHeading
+        :level="3"
+        title="Choose language"
+        class="container__lang--title"
+      />
+      <SfList>
+        <SfListItem v-for="lang in availableLocales" :key="lang.code">
+          <a :href="switchLocalePath(lang.code)">
+            <SfCharacteristic class="language">
+              <template #title>
+                <span>{{ lang.label }}</span>
+              </template>
+              <template #icon>
+                <SfImage :src="`/icons/langs/${lang.code}.webp`" width="20" alt="Flag" class="language__flag" />
+              </template>
+            </SfCharacteristic>
+          </a>
+        </SfListItem>
+      </SfList>
+    </SfBottomModal>
+  </div>
+</template>
+
+<script>
+import {
+  SfBottomModal,
+  SfButton,
+  SfCharacteristic,
+  SfHeading,
+  SfImage,
+  SfList,
+  SfSelect
+} from '@storefront-ui/vue';
+import { onSSR } from '@vue-storefront/core';
+import { useStore, storeGetters } from '@vue-storefront/commercetools';
+import { ref, computed } from '@vue/composition-api';
+
+export default {
+  components: {
+    SfBottomModal,
+    SfButton,
+    SfCharacteristic,
+    SfHeading,
+    SfImage,
+    SfList,
+    SfSelect
+  },
+  setup(props, context) {
+    const { locales, locale, defaultLocale } = context.root.$i18n;
+    const { load, change, response } = useStore();
+    const isLangModalOpen = ref(false);
+    const availableLocales = computed(() => locales.filter(i => i.code !== locale));
+
+    onSSR(async () => {
+      await load();
+    });
+
+    const availableStores = computed(() => storeGetters.getItems(response.value));
+    const selectedStore = computed(() => storeGetters.getSelected(response.value));
+    const changeStore = async (store) => {
+      isLangModalOpen.value = false;
+      await change({store});
+    };
+    const isStoreSelected = (store) => selectedStore.value?.id === store.id;
+    const getStoreLocale = (store) => store?.locales[0] ?? defaultLocale;
+
+    return {
+      load,
+      changeStore,
+      response,
+      availableStores,
+      selectedStore,
+      isStoreSelected,
+      getStoreLocale,
+      //
```

Comment:
Do we need this?

Reply:
those 3 are being used in the theme section :)